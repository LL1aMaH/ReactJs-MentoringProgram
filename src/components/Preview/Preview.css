.preview {
  width: 100%;
  height: 25rem;
  display: flex;
  flex-direction: column;
  position: relative;
}

.buttons {
  width: 100%;
  display: flex;
  justify-content: space-between;
  padding: 1rem 5rem;
}

.button {
  backdrop-filter: blur(10px);
  padding: 1rem;
  color: var(--primary-link-color);
  font-size: 18px;
}

.previewContent {
  position: absolute;
  top: 0;
  width: 100vw;

  & p {
    color: var(--primary-font-color);
    font-size: 30px;
    margin-top: 3rem;
    padding-left: 10rem;
  }
}

.searchBox {
  padding: 2rem 10rem;
}
