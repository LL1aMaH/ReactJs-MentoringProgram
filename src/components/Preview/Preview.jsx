import React, { memo, useState } from 'react';

import { Logo } from '../Logo/Logo';

import { SuccessfulAddMovieModal } from '../Modals/SuccessfulAddMovieModal/SuccessfulAddMovieModal';

import styles from '../../../styles/Preview.module.scss'

// import { AddMovieModal } from '../Modals/AddMovieModal/AddMovieModal';
import { SearchBox } from './components/SearchBox/SearchBox';


export const Preview = memo(function Preview({ searchValue }) {
  const [showSuccessfulAddMovieModal, setShowSuccessfulAddMovieModal] = useState(false);
  return (
    <>
      <div
        className={styles.preview}
        style={{
          backgroundImage: "url('/cover-image.jpg')",
          filter: 'blur(5px)',
          backgroundSize: "cover"
        }}
      />
      <div className={styles.previewContent}>
        <div className={styles.buttons}>
          <Logo />
          {/* <AddMovieModal
            className={styles.button}
            setShowSuccessfulAddMovieModal={setShowSuccessfulAddMovieModal}
          /> */}
          {showSuccessfulAddMovieModal && <SuccessfulAddMovieModal />}
        </div>
        <p>FIND YOUR MOVIE</p>
        <SearchBox className={styles.searchBox} searchValue={searchValue} />
      </div>
    </>
  );
});
