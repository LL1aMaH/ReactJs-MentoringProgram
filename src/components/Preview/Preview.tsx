import React, { memo, useState } from 'react';

import { Logo } from 'Components/Logo';

import { AddMovieModal } from '../Modals/AddMovieModal';
import { SearchBox } from './components/SearchBox';

import styles from './Preview.css';

export const Preview = memo(function Preview(): JSX.Element {
  const [state, setState] = useState<string[]>([]);

  const handleClick = (item: string[]) => {
    setState([...item]);
  };
  return (
    <>
      <div
        className={styles.preview}
        style={{
          backgroundImage: 'url(src/Assets/pictures/cover-image.jpg)',
          filter: 'blur(5px)',
        }}
      />
      <div className={styles.previewContent}>
        <div className={styles.buttons}>
          <Logo />
          <AddMovieModal className={styles.button} handleClick={handleClick} value={state} />
        </div>
        <p>FIND YOUR MOVIE</p>
        <SearchBox className={styles.searchBox} />
      </div>
    </>
  );
});
