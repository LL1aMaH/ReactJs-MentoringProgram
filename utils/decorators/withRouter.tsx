import React from 'react';

import { MemoryRouter } from 'react-router-dom';

export type TComponentFunc = () => JSX.Element;
export type TDecoratorLocation = {
  pathname?: string;
  search?: string;
};

const withRouter = (
  componentFunc: TComponentFunc,
  location: TDecoratorLocation = { pathname: '/' },
): JSX.Element => {
  return <MemoryRouter initialEntries={[location]}>{componentFunc()}</MemoryRouter>;
};

export default withRouter;
