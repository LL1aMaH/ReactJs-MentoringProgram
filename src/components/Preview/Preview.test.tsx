import React from 'react';
import { render } from '@testing-library/react';

import withRouter from '../../../utils/decorators/withRouter';

import { Preview } from './Preview';

describe('Preview', () => {
  it('renders correctly', () => {
    const component = render(withRouter(() => <Preview />));
    expect(component.asFragment()).toMatchSnapshot();
  });
});
