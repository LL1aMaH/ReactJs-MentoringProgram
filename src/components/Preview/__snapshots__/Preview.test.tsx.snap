// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview renders correctly 1`] = `
<DocumentFragment>
  <div
    class="preview"
    style="background-image: url(/src/Assets/pictures/cover-image.jpg); filter: blur(5px);"
  />
  <div
    class="previewContent"
  >
    <div
      class="buttons"
    >
      <a
        class="logo"
        href="/"
      >
        <b>
          netflix
        </b>
        roulette
      </a>
      <button
        class="button button"
        data-testid="addMovie"
        tabindex="0"
      >
        + ADD MOVIE
      </button>
    </div>
    <p>
      FIND YOUR MOVIE
    </p>
    <div
      class="searchBox searchBox"
    >
      <input
        placeholder="What do you want to watch?"
        value=""
      />
      <button
        class="button button"
        data-testid="button"
        tabindex="0"
      >
        SEARCH
      </button>
    </div>
  </div>
</DocumentFragment>
`;
